lib/temporal.explorer.ts

```diff
@@ -29,40 +29,30 @@ export class TemporalExplorer
   @Inject(TEMPORAL_MODULE_OPTIONS_TOKEN) private options: TemporalModuleOptions;
   private readonly logger = new Logger(TemporalExplorer.name);
   private worker: Worker;
-  private timerId: ReturnType<typeof setInterval>;
+  private workerRunPromise: Promise<void>
 
   constructor(
     private readonly discoveryService: DiscoveryService,
     private readonly metadataAccessor: TemporalMetadataAccessor,
     private readonly metadataScanner: MetadataScanner,
   ) {}
 
-  clearInterval() {
-    this.timerId && clearInterval(this.timerId);
-    this.timerId = null;
-  }
-
   async onModuleInit() {
     await this.explore();
   }
 
-  onModuleDestroy() {
+  async onModuleDestroy() {
     try {
       this.worker?.shutdown();
+      await this.workerRunPromise;
+
     } catch (err: any) {
       this.logger.warn('Temporal worker was not cleanly shutdown.', { err });
     }
-
-    this.clearInterval();
   }
 
   onApplicationBootstrap() {
-    this.timerId = setInterval(() => {
-      if (this.worker) {
-        this.worker.run();
-        this.clearInterval();
-      }
-    }, 1000);
+    this.workerRunPromise = this.worker?.run();
   }
```

Comment:
Per https://docs.nestjs.com/fundamentals/lifecycle-events, `onModuleInit` is called before `onApplicationBootstrap`, and no other code calls `.explore` in this repo, so I don't *think* it can be set at a later time with any of the publicly described interfaces, but I can reinstate this timer if it is needed for some reason